service/src/main/java/org/cbioportal/service/impl/GenericAssayServiceImpl.java

```diff
@@ -110,16 +103,15 @@
     List<String> sampleIds, List<String> genericAssayStableIds, String projection) throws MolecularProfileNotFoundException {
         List<GenericAssayData> result = new ArrayList<>();
 
-        List<String> distinctMolecularProfileIds = molecularProfileIds.stream().distinct().sorted().collect(Collectors.toList());
+        SortedSet<String> distinctMolecularProfileIds = new TreeSet<>(molecularProfileIds);
```

Comment:
Any specific reason to use `SortedSet`?

Reply:
I use `SortedSet` since the previous `List` was being `.sorted()` before being collected, so I want to preserve that ordering in my `Set` implementation.